app/com/lynxanalytics/biggraph/graph_operations/SQLFiltering.scala

```diff
@@ -0,0 +1,58 @@
+// Operations for the "Filter with SQL" frontend operation.
+package com.lynxanalytics.biggraph.graph_operations
+
+import com.lynxanalytics.biggraph.graph_api._
+import com.lynxanalytics.biggraph.spark_util.Implicits._
+
+object IDTableToVertexSet extends OpFromJson {
+  class Input extends MagicInputSignature {
+    val t = table
+  }
+  class Output(implicit instance: MetaGraphOperationInstance) extends MagicOutput(instance) {
+    val vs = vertexSet
+  }
+  def fromJson(j: JsValue) = IDTableToVertexSet((j \ "column").as[String])
+}
+case class IDTableToVertexSet(column: String)
+  extends SparkOperation[IDTableToVertexSet.Input, IDTableToVertexSet.Output] {
+  override val isHeavy = true
+  @transient override lazy val inputs = new IDTableToVertexSet.Input
+  def outputMeta(instance: MetaGraphOperationInstance) = new IDTableToVertexSet.Output()(instance)
+  override def toJson = Json.obj("column" -> column)
+  def execute(
+    inputDatas: DataSet,
+    o: IDTableToVertexSet.Output,
+    output: OutputBuilder,
+    rc: RuntimeContext): Unit = {
+    implicit val id = inputDatas
+    val partitioner = rc.partitionerForNRows(inputs.t.data.df.count)
+    val vs = inputs.t.data.df.select(column).rdd.map(row => (row.getLong(0), ()))
+    output(o.vs, vs.sortUnique(partitioner))
+  }
+}
+
+object EmbeddingByID extends OpFromJson {
+  class Input extends MagicInputSignature {
+    val original = vertexSet
+    val filtered = vertexSet
+  }
+  class Output(implicit instance: MetaGraphOperationInstance, inputs: Input) extends MagicOutput(instance) {
+    val identity = edgeBundle(inputs.filtered.entity, inputs.original.entity, EdgeBundleProperties.embedding)
+  }
```

Comment:
Maybe we have this somewhere already? Seems like such an obvious operation.

Reply:
It's basically VertexSetIntersection. Two differences: that one does some work to really just keep the common ids and the result is partitioned as the original in that case. I'm not hundred percent sure whether these are good or bad. :) But we use that for attribute filters, so maybe safest to use it here as well?

Reply:
An alternative would be if a single operation would do the whole job. Input: a VertexSet and an id table. Output: the new vertex set and its embedding. I don't see how creating a VertexSet out of nothing from ids will be useful on its own. And then it's nicer not to create the embedding after the fact. WDYT?

Reply:
During breakfast I thought of a better argument for using a single operation than "It feels nicer to me". In the current two operations setup, a new partitioner is pulled out of thin air, it will be always different than the original, so everything will have to be repartitioned. If this new partitioner is very different, then it's fine. But if it's just slightly different, e.g. we went from 1M to 900k vertices, then it would be better to keep the original partitioner. If on the other hand we use the partitioner of the original VS, then exactly the right thing will happen: if the difference is small, we keep it. If the difference is big then the data manager will realize and repartition.

Reply:
> During breakfast I thought of a better argument for using a single operation than "It feels nicer to me".

Good points, thanks. Done.